packages/core/src/tooltip/Tooltip.tsx

```diff
@@ -137,49 +128,21 @@ export const Tooltip = forwardRef<HTMLDivElement, TooltipProps>(
             ref: triggerRef,
           })}
 
-        {open && !disabled && (
-          <FloatingPortal>
-            {/* The provider is needed to support the use case where an app has nested modes. The element that is portalled needs to have the same style as the current scope */}
-            <SaltProvider>
-              <div
-                className={clsx(
-                  withBaseName(),
-                  withBaseName(status),
-                  className
-                )}
-                ref={floatingRef}
-                {...getTooltipProps()}
-              >
-                <div className={withBaseName("container")}>
-                  {!hideIcon && (
-                    <StatusIndicator
-                      status={status}
-                      size={1}
-                      className={withBaseName("icon")}
-                    />
-                  )}
-                  <span
-                    id={a11yProps?.["aria-describedby"]}
-                    className={withBaseName("content")}
-                  >
-                    {content}
-                  </span>
-                </div>
-                {!hideArrow && (
-                  <FloatingArrow
-                    {...arrowProps}
-                    className={withBaseName("arrow")}
-                    strokeWidth={1}
-                    fill="var(--salt-container-primary-background)"
-                    stroke="var(--tooltip-status-borderColor)"
-                    height={5}
-                    width={10}
-                  />
-                )}
-              </div>
-            </SaltProvider>
-          </FloatingPortal>
-        )}
+        <FloatingComponent
```

Comment:
I feel like the API for this component is huge, have you considered just moving Floating UI into Floating Component and allow the user to completely swap out the component?

In my mind we would just pass the trigger Ref or element to the component and the open/closed state

Reply:
I'm not sure that works:

1) The same FloatingComponent is used for each of our components (Tooltip/Dropdown etc), but they have different inputs into useFloatingUI e.g. different middleware/interactions etc. So we need to call/configure floating ui from outside the FloatingComponent

2) Our components sometimes use the output of useFloatingUi from outside of the FloatingComponent e.g. Tooltip uses arrowProps

3) This would mean consumers having to call/configure useFloatingUI themselves when they pass a custom FloatingComponent, which feels like something we've tried to abstract away from them

In addition I would advise against making major changes to the API at this point. As I understand it there are very few consumers who will be using this API for very specific use-cases. This PR has gone through several iterations of these consumers testing out the solution and us tweaking it to make sure it works for their use-case. 

At this point the solution/API works for them, and unless there is an impact on the general use of these components I see little value in significant re-work causing another cycle of iterations and blocking these consumers.